game.js

```diff
@@ -547,69 +547,60 @@ module.exports = function createGame(options) {
         return lodash.intersection(state.roles, lodash.flatten([roles]))[0];
     }
 
-    function command(playerIdx, command) {
-        debug('command from player: ' + playerIdx);
-        debug(command);
-        var i, action, message;
-        var playerState = state.players[playerIdx];
-        if (playerState == null) {
-            throw new GameException('Unknown player');
+    const playerIsReady = (command, playerIdx) => {
+        if (state.players[playerIdx].isReady !== true) {
+            throw new GameException('You cannot start the game');
         }
-        if (command.command == 'leave') {
-            // You can always leave, even if your state id is old.
-            playerLeft(playerIdx);
+    };
+    const playerIsUndead = (command, playerIdx) => {
+        if (state.players[playerIdx].influenceCount == 0) {
+            throw new GameException('Dead players cannot challenge');
         }
-        else if (command.stateId != state.stateId) {
-            throw new GameException('Stale state (' + command.stateId + '!=' + state.stateId + ')');
+    };
+    const playersTurn = (command, playerIdx) => {
+        if (state.state.playerIdx != playerIdx) {
+            throw new GameException('Not your turn');
         }
-        else if (command.command == 'start') {
-            if (playerState.isReady !== true) {
-                throw new GameException('You cannot start the game');
-            }
-            start(command.gameType);
-
-        } else if (command.command == 'ready') {
-            if (state.state.name != stateNames.WAITING_FOR_PLAYERS) {
-                throw new GameException('Incorrect state');
-            }
-            playerReady(playerIdx);
-
-        } else if (command.command == 'add-ai') {
-            if (state.state.name != stateNames.WAITING_FOR_PLAYERS) {
-                throw new GameException('Incorrect state');
-            }
-            if (playerState.isReady !== true) {
-                throw new GameException('You cannot add AI players');
-            }
-            addAiPlayer();
-
-        } else if (command.command == 'remove-ai') {
-            if (state.state.name != stateNames.WAITING_FOR_PLAYERS) {
-                throw new GameException('Incorrect state');
-            }
-            if (playerState.isReady !== true) {
-                throw new GameException('You cannot remove AI players');
-            }
-            removeAiPlayer();
-
-        } else if (command.command == 'play-action') {
-            if (state.state.name != stateNames.START_OF_TURN) {
-                throw new GameException('Incorrect state');
-            }
-            if (state.state.playerIdx != playerIdx) {
-                throw new GameException('Not your turn');
-            }
-            action = actions[command.action];
+    };
+    const invalidPlayer = (stateField) => (command, playerIdx) => {
```

Comment:
For clarity, rename to something like `checkPlayerDoesNotActOnSelf`, and the message should end in `on himself` or similar.